src/bookKeeping.C

```diff
@@ -348,6 +349,198 @@ void MeshBlock::processDonors(HOLEMAP *holemap, int nmesh, int **donorRecords,do
+          }
+        }
+      }
+    }
+  }
+
+  for(i=0;i<nwbc;i++){
+    if(iblank[wbcnode[i]-BASE]==0){
+      fprintf(stderr,"--------------------------------------------------------------------\n");
+      fprintf(stderr,"Alarm from process %d : wall node is being tagged as a hole %d %p\n",
+                      myid,wbcnode[i]-BASE,donorList[wbcnode[i]-BASE]);
+      ii=wbcnode[i]-BASE;
+      fprintf(stderr,"xloc=%e %e %e\n",x[3*ii],x[3*ii+1],x[3*ii+2]);
+      fprintf(stderr,"Computations will continue, but may suffer from accuracy problems\n");
+      fprintf(stderr,"Please recheck positions of your grids\n");
+      fprintf(stderr,"--------------------------------------------------------------------\n");
+    }
+  }
+
+  /* ===================================================================== */
+  /* 2. mark mandatory fringes as neighbors (nfringe depth) of hole points */
+  /* ===================================================================== */
+  mtag  = (int *) malloc(sizeof(int)*nnodes);
+  mtag1 = (int *) malloc(sizeof(int)*nnodes);
+
+  for(i=0;i<nnodes;i++){
+    mtag[i] = mtag1[i] = 0;
+    if(iblank[i]==0) mtag[i] = mtag1[i] = 1;
+  }
+
+  for(iter=0;iter<nfringe;iter++){
+    for(n=0;n<ntypes;n++){
+      nvert=nv[n];
+      for(i=0;i<nc[n];i++){
+	for(m=0;m<nvert;m++){
+          if(mtag[(vconn[n][nvert*i+m]-BASE)]==1){
+            for(mm=0;mm<nvert;mm++){
+              if(m!=mm && mtag[vconn[n][nvert*i+mm]-BASE] !=1){
+                mtag1[vconn[n][nvert*i+mm]-BASE] = 1;
+              }
+            }
+          }
+        }
+      }
+    }
+    for(i=0;i<nnodes;i++) mtag[i] = mtag1[i];
+  }
+
+  for(i=0;i<nnodes;i++){
+    if(mtag1[i] && iblank[i]) nodeRes[i]=BIGVALUE;
+  }
+  TIOGA_FREE(mtag);
+  TIOGA_FREE(mtag1);
+
+  /* =============== */
+  /* 3. find fringes */
+  /* =============== */
+  *nrecords=0;
+  for(i=0;i<nnodes;i++){
+    verbose=0;
+    if(verbose){
+      TRACEI(i);
+      TRACEI(iblank[i]);
+    }
+
+    if(donorList[i]!=NULL && iblank[i]!=0){
+      temp=donorList[i];
+      if(verbose) TRACED(nodeRes[i]);
+
+      while(temp!=NULL){
+        if(verbose) TRACED(temp->donorRes);
+
+        if(temp->donorRes < nodeRes[i]){
+          iblank[i]=-temp->donorData[1];
+
+          if(verbose){
+            TRACEI(iblank[i]);
+            TRACEI(temp->donorData[0]);
+            TRACEI(temp->donorData[1]);
+            TRACEI(temp->donorData[2]);
+          }
+          (*nrecords)++;
+          break;
+        }
+        temp=temp->next;
+      }
+    }
+  }
+
+  /* ==================================================== */
+  /* 4. set the records to send back to the donor process */
+  /* ==================================================== */
+  (*donorRecords)=(int *)malloc(sizeof(int)*3*(*nrecords));
+  (*receptorResolution)=(double *)malloc(sizeof(double)*(*nrecords));
+
+  m=k=0;
+  for(i=0;i<nnodes;i++){
+    verbose=0;
+    if(iblank[i] < 0){
+      temp=donorList[i];
+      while(temp!=NULL){
```

Comment:
Does this need to exist?

Reply:
It's a hold-over from the original function lines 320-330, so I wouldn't change the functionality.

Reply:
I'm wondering if there's a bug in the original function. My guess is there should be
'''temp=temp->next;'''
Inside the while loop. Thoughts ?

Reply:
it doesn't make any sense without some kind of increment inside the `while`